From 5e79fb0fa44fa48f78a0ab146d255b22372e0b9e Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:41 -0400
Subject: [cpufreq] pcc: don't initialize part of policy set by core

Message-id: <1409235879-1283-68-git-send-email-prarit@redhat.com>
Patchwork-id: 89398
O-Subject: [RHEL7.1 PATCH BZ 1134639 067/185] cpufreq: pcc: don't initialize part of policy set by core
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 6b67ca322d409003c29e119f8c8beeecd5785e20
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Thu Oct 3 20:28:53 2013 +0530

    cpufreq: pcc: don't initialize part of policy set by core

    Many common initializations of struct policy are moved to core now and hence
    this driver doesn't need to do it. This patch removes such code.

    Most recent of those changes is to call ->get() in the core after calling
    ->init().

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/pcc-cpufreq.c b/drivers/cpufreq/pcc-cpufreq.c
index f4e5dde..1a306ce 100644
--- a/drivers/cpufreq/pcc-cpufreq.c
+++ b/drivers/cpufreq/pcc-cpufreq.c
@@ -558,13 +558,6 @@ static int pcc_cpufreq_cpu_init(struct cpufreq_policy *policy)
   ioread32(&pcch_hdr->nominal) * 1000;
  policy->min = policy->cpuinfo.min_freq =
   ioread32(&pcch_hdr->minimum_frequency) * 1000;
- policy->cur = pcc_get_freq(cpu);
-
- if (!policy->cur) {
-  pr_debug("init: Unable to get current CPU frequency\n");
-  result = -EINVAL;
-  goto out;
- }
 
  pr_debug("init: policy->max is %d, policy->min is %d\n",
   policy->max, policy->min);
-- 
1.7.1